quartz/components/scripts/search.inline.ts

```diff
@@ -248,6 +288,42 @@ document.addEventListener("nav", async (e: unknown) => {
     }
   }
 
+  async function fetchContent(slug: FullSlug): Promise<Element[] | undefined> {
```

Comment:
```suggestion
  async function fetchContent(slug: FullSlug): Promise<Element[]> {
```

let's just reject instead of resolving `undefined`